chore: Split test and lint scripts

karfau opened the pull request:
To fix #111 I think it's good enough to split test and lint into two different build steps.
Separate actions are not required since the github actions details view now shows what steps failed.
![image](https://user-images.githubusercontent.com/135657/130516059-1dced967-0721-46a1-bf36-30359d968e88.png)


To be able to do that split, I needed to drop the execution of the `lint` script from the `test` script. The `start` script still runs both in watch mode, so it's still easy to run both when developing.

brodybits reviewed (commented):
It looks to me like npm-run-all is no longer needed and should be removed ?

karfau commented:
@brodybits you are right, I dropped it, quite some reduction on devDependencies side
```
$ npm rm npm-run-all
removed 49 packages, updated 5 packages and audited 600 packages in 12.223s
```

brodybits commented on .github/workflows/test-node.js.yml:
```diff
@@ -28,4 +28,6 @@ jobs:
       with:
         node-version: ${{ matrix.node-version }}
-    - run: npm cit
+    - run: npm ci --no-audit
```
minor question: Can you quickly describe the motivation behind adding `--no-audit` here?

karfau replied:
Yes:
a) (Since we have tools to provide update PRs) nobody is looking at the audit output on CI
b) [it saves time to not do it](http://www.tiernok.com/posts/2019/faster-npm-installs-during-ci/)

brodybits reviewed (approved):
LGTM overall

brodybits commented:
@karfau it looks like this broke the Stryker run in master ... definitely not the first time we hit this kind of thing.

I am wondering if using Stryker Jest runner, as I tried in PR #208, could help avoid this failure.

I am thinking now if we could find a way to add another task to check that Stryker can do its initial run in the future PRs?

karfau commented:
> I am thinking now if we could find a way to add another task to check that Stryker can do its initial run in the future PRs?

Yes, also thought about this, but I couldn't find any documentation about such a "dryrun" mode yet.
